src/api/roles/client/client.py

```
@@ -557,3 +589,150 @@ def get_my_coach_requests(db = Depends(get_session), acc: Account = Depends(get_
+
+    return [
+        {
+            "date": pic.date,
+            "progress_pic_url": pic.progress_pic_url,
+        }
+        for pic in pictures
+    ]
+
+
+@router.get("/my_coach")
+def get_my_coach(db = Depends(get_session), acc: Account = Depends(get_client_account)):
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)